src/settingsvpnmodel.cpp

```diff
@@ -52,7 +52,8 @@ const auto legacyDefaultDomain(QStringLiteral("merproject.org"));
 int numericValue(VpnConnection::ConnectionState state)
 {
     return (state == VpnConnection::Ready ? 3 :
-                (state == VpnConnection::Configuration ? 2 : 0));
+                (state == VpnConnection::Configuration ? 2 :
+                (state == VpnConnection::Association ? 1 : 0)));
 }
```

Comment:
While this gets the job done and follows the style already in use, the interestingly indented superfluous parentheses within recursive ternary ops and all make me yearn for a 'switch' ...

Reply:
I wonder if the numeic value could be just the value of the enum instead.

Reply:
I think that this had something to do with the case that we account for only the relevant states. Disconnect seems to be the last, i.e., highest in numeric value in [vpnconnection.h](https://github.com/sailfishos/libconnman-qt/blob/master/libconnman-qt/vpnconnection.h). There is elsewhere in the UI some bits about best state, it may be related.